pkg/kubelet/dockertools/docker_manager.go

```diff
@@ -1586,19 +1586,16 @@ func containerAndPodFromLabels(inspect *dockertypes.ContainerJSON) (pod *api.Pod
 			if container == nil {
 				err = fmt.Errorf("unable to find container %s in pod %v", name, pod)
 			}
-		} else {
-			pod = nil
+			return
```

Comment:
Shouldn't you break the logic here by returning earlier without running the following logic?